File: Arcade/TheCore/Javascript/comfortableNumbers.js
function s(n) {
    return [...n+""].map(i => +i).reduce((t, a) => t + a)
}

function comfortableNumbers(l, r) {
    let pairs = 0;

    for (let i = l; i < r; i++)
        for (let j = i + 1; j <= r; j++)
            if (i >= j - s(j) && i <= j + s(j)
                && j >= i - s(i) && j <= i + s(i)) pairs++;
    
    return pairs;
}

File: Arcade/TheCore/Kotlin/comfortableNumbers.kt
fun comfortableNumbers(l: Int, r: Int): Int {
    fun Int.s(): Int = (""+this).sumOf { (""+it).toInt() };
    
    var pairs: Int = 0;
    
    (l..r).forEach { i -> 
        ((i + 1)..r).forEach { j -> 
            if (i in ((j - j.s())..(j + j.s()))
                && j in ((i - i.s())..(i + i.s()))) pairs++;
        }
    };
    
    return pairs;
}